### ISSUE TITLE
Warn on omitted commas in match arms

### ISSUE THREAD

**Issue by kangalioo** [rust-lang/rust-clippy#6445]
### What it does

Rust allows you to omit the separating commas when matching on struct-like enum variants:

```rust
match enum_value {
    Variant1 => {
        // ...
    }
    Variant2 => {
        // ...
    }
    Variant3 => {
        // ...
    }
}
```

This is inconsistent, because Rust doesn't allow the commas in any other part of the language, as far as I know. This lint will warn when that "feature" is being used.

### Categories (optional)

- Kind: clippy::pedantic, maybe clippy::style

*What is the advantage of the recommended code over the original code*

Using commas there is consistent with the rest of the language. Also, it ensures that you don't accidentally miss commas in _some_ of the variants, which makes it look ugly:

```rust
match enum_value {
    Variant1 => {
        // ...
    },
    Variant2 => {
        // ...
    }
    Variant3 => {
        // ...
    },
}
```

### Drawbacks

Some users might prefer not to have commas there.

### Example

```rust
match enum_value {
    Variant1 => {
        // ...
    }
    Variant2 => {
        // ...
    }
    Variant3 => {
        // ...
    }
}
```

Could be written as:

```rust
match enum_value {
    Variant1 => {
        // ...
    },
    Variant2 => {
        // ...
    },
    Variant3 => {
        // ...
    },
}
```

**Comment by ebroto** [comment]
Hi @kangalioo, thanks for your interest in improving Clippy.

This is definitely `rustfmt` territory. Clippy does not have the tools to enforce this kind of stylistic rules.
